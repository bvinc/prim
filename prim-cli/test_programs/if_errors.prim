fn main() {
    let x = 42
    if 1 == 0 {
        println(x)
    }
    // This should cause a type error - condition must be boolean
    if x {
        println(x)
    }
}
