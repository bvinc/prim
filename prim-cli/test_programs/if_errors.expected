ERROR: Type check error: Type mismatch: expected bool, found i64
